src/main/resources/org/jpeek/metrics/CCM.xsl

```diff
@@ -0,0 +1,118 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0">
+  <xsl:param name="ctors" select="0"/>
+  <xsl:template match="skeleton">
+    <metric>
+      <xsl:apply-templates select="@*"/>
+      <title>CCM</title>
+      <description>
+        <xsl:text>Class Connection Metric (CCM) is based on the connection
+          graph of a class where each node is a method and there exists an
+          "edge" or connection between nodes if either both nodes access the
+          same attribute of a class, or if both call another method of the
+          class. The formula is defined as CCM = NC/(NMP x NCC), where
+          NC = number of connections according to the previous criteria,
+          NMP = maximum number of possible connections (NMP = (N x (N - 1))/2),
+          and NCC = number of connected nodes.</xsl:text>
+      </description>
+      <xsl:apply-templates select="node()"/>
+    </metric>
+  </xsl:template>
+  <xsl:template match="class">
+    <xsl:variable name="methods" select="methods/method[($ctors=0 and @ctor='false') or $ctors=1]"/>
+    <xsl:variable name="edges">
+      <xsl:for-each select="$methods">
+        <xsl:variable name="method" select="."/>
+        <xsl:for-each select="$method/following-sibling::method">
+          <xsl:variable name="other" select="."/>
+          <!--
+           @todo #66:30min For CCM, the following test cases need to be
+            added after #103 is fixed: NoMethods, WithoutAttributes,
+            OneMethodCreatesLambda, OneVoidMethodWithoutParams. The CCM value
+            for these cases is NaN.
+          -->
+          <!--
+           @todo #66:30min For CCM, cannot add test case for class Bar because
+            it is blocked by #114. Omission of the 'NAME' <op> for "getKey"
+            messes up calculations. Add test case after #114 is fixed.
+          -->
+          <!--
+           @todo #66:30min For CCM, need to add OR condition to the following
+            xsl:if to check whether two methods both call another method in
```

Comment:
@llorllale I understand from this puzzle that the metric won't execute properly because you don't have the ``OR`` condition. If it is true, then should we delcare this metric as a Report in APP yet? Won't the app crash if a class which requires that ``OR`` operation is analysed?

Reply:
@amihaiemil no, the app won't crash, it would just output incorrect values for CCM

Reply:
@llorllale hmm.. then I think maybe we should not declare it in the Report yet. Because, as usual, the app may be released before all these puzzles are fixed, so we need to always try and keep a most consistent state of the app. 

I would say remove the metric from the App report and leave a puzzle (but only 15min estimation, ``@todo #66:15min Add the CCM metric here when all the puzzles about it are resolved``. 

What do you think? If we release jpeek before all this is solved, the users may rely on this report and it will be incorrect.

Reply:
@amihaiemil i defer to your judgement since you're more familiar with yegor's process and methods. I'll do as you suggested and let you know